pkg/actuators/machine/reconciler_test.go

```diff
@@ -900,3 +900,233 @@ func TestGetMachineInstances(t *testing.T) {
 		})
 	}
 }
+
+func TestDeletePlacementGroup(t *testing.T) {
+	instanceID := "i-02fa4197109214b46"
+	imageID := "ami-a9acbbd6"
+	pgName := "mapi-placement-group"
+
+	machine, err := stubMachine()
+	if err != nil {
+		t.Fatalf("unable to build stub machine: %v", err)
+	}
+
+	machinePc, err := ProviderSpecFromRawExtension(machine.Spec.ProviderSpec.Value)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	machinePc.Placement.GroupName = pgName
+
+	config, err := RawExtensionFromProviderSpec(machinePc)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	machine.Spec.ProviderSpec = machinev1.ProviderSpec{Value: config}
+
+	awsCredentialsSecret := stubAwsCredentialsSecret()
+	userDataSecret := stubUserDataSecret()
+
+	testCases := []struct {
+		testcase       string
+		providerStatus machinev1.AWSMachineProviderStatus
+		awsClientFunc  func(*gomock.Controller) awsclient.Client
+		expectError    bool
+		expectDeleted  bool
+	}{
+		{
+			testcase: "Successful deletion",
+			providerStatus: machinev1.AWSMachineProviderStatus{
+				InstanceID: aws.String(instanceID),
+			},
+			awsClientFunc: func(ctrl *gomock.Controller) awsclient.Client {
+				mockAWSClient := mockaws.NewMockClient(ctrl)
+
+				request := &ec2.DescribePlacementGroupsInput{
+					GroupNames: []*string{aws.String(pgName)},
+				}
```

Comment:
A lot of this is duplicated across the tests, maybe we can set this all up as part of the test rather than the test case?